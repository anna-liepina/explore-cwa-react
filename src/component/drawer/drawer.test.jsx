import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, act } from '@testing-library/react';
import Drawer from './drawer';

describe('<Drawer/>', () => {
    const props = {
        onClose: jest.fn(),
    };

    const optionalProps = {
        'data-cy': 'optionalProps.data-cy',
        className: 'optionalProps.className',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<Drawer {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<Drawer {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('callbacks', () => {
        beforeEach(() => {
            jest.useFakeTimers();
            jest.clearAllTimers();
        });

        it('onClose', () => {
            const { container } = render(<Drawer {...props} />);

            act(() => {
                fireEvent.click(container.querySelector('[data-cy="--drawer--close"]'));

                jest.runAllTimers();
            });

            expect(props.onClose).toBeCalled();
        });
    });
});
