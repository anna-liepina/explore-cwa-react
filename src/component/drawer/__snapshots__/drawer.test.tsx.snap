// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> render with default/required props 1`] = `
<DocumentFragment>
  <div
    class="drawer_overlay"
  >
    <aside
      class="drawer--open"
      data-cy="--drawer"
      style="transition-duration: 0.25s;"
    >
      <header
        class="drawer_head"
      >
        <button
          class="drawer_button--close"
          data-cy="--drawer--close"
        >
          ×
        </button>
      </header>
      <section
        class="drawer_content"
      >
        <div
          data-cy="drawer-child"
        />
      </section>
    </aside>
  </div>
</DocumentFragment>
`;

exports[`<Drawer/> render with optional/required props 1`] = `
<DocumentFragment>
  <div
    class="drawer_overlay"
  >
    <aside
      class="optionalProps.className drawer--open"
      data-cy="optionalProps.data-cy--drawer"
      style="transition-duration: 0.25s;"
    >
      <header
        class="drawer_head"
      >
        <button
          class="drawer_button--close"
          data-cy="optionalProps.data-cy--drawer--close"
        >
          ×
        </button>
      </header>
      <section
        class="drawer_content"
      >
        <div
          data-cy="drawer-child"
        />
      </section>
    </aside>
  </div>
</DocumentFragment>
`;
